### PR TITLE
Depend on rocm-core package

### PR DESCRIPTION
**Pull request by icarus-sparry**
Add dependencies on the new rocm-core package by default.
Disable by setting ROCM_DEP_ROCMCORE to FALSE

### CODE REVIEW

**Review by lawruble13 — COMMENTED** [review]
Is the `rocm-core` package available for previous versions of ROCm? If not, this may cause issues for CI, so the flag will need to be added to the CI systems. I know some of the math libraries test from the current clients package.

File: share/rocm/cmake/ROCMCreatePackage.cmake [hunk 1]
```diff
@@ -36,6 +38,16 @@ else()
     endfunction()
 endif()
 
+macro(add_rocm_core_dependency)
+    # Optionally add depenency on rocm-core
+    # This mainly empty package exists to allow all of rocm
+    # to be removed in one step by removing rocm-core
+    if(ROCM_DEP_ROCMCORE)
+        rocm_join_if_set(", " CPACK_DEBIAN_PACKAGE_DEPENDS "rocm-core")
```
Comment (lawruble13) [inline share/rocm/cmake/ROCMCreatePackage.cmake:46]: When #58 is merged in, these can be replaced with `rocm_package_add_dependencies("rocm-core")`.

Comment (icarus-sparry) [reply share/rocm/cmake/ROCMCreatePackage.cmake:46]: Ack.
Do you want me to wait for #58 and update this PR?

File: share/rocm/cmake/ROCMCreatePackage.cmake [hunk 2]
```diff
@@ -36,6 +38,16 @@ else()
     endfunction()
 endif()
 
+macro(add_rocm_core_dependency)
```
Comment (icarus-sparry) [inline share/rocm/cmake/ROCMCreatePackage.cmake:41]: This is a macro to keep the linter happy. It was objecting to the complexity of the code.

Comment (pfultz2) [reply share/rocm/cmake/ROCMCreatePackage.cmake:41]: It should be prefixed with `rocm_package_`.

File: share/rocm/cmake/ROCMCreatePackage.cmake (hunk 1 above)
Comment (lawruble13) [reply share/rocm/cmake/ROCMCreatePackage.cmake:46]: Nah, #58 still has some work to be done, I'll make the change to it after this is merged in.

**Review by pfultz2 — APPROVED** [review]

**Review by lawruble13 — APPROVED** [review]

**Comment by icarus-sparry** [comment]
@pfultz2 Is there anything else required before this can be merged?

**Comment by saadrahim** [comment]
@pfultz2 I added a changelog earlier that @lawruble13 has kept updated. Should this type of change be documented in it? (I was away until today and did not get a chance to comment earlier).

**Comment by icarus-sparry** [comment]
> @pfultz2 I added a changelog earlier that @lawruble13 has kept updated. Should this type of change be documented in it? (I was away until today and did not get a chance to comment earlier).

I am not a big fan of keeping documentation on changes in 2 different places. We already have the git log of commit messages. There are several projects which create a ChangeLog out of commit messages.

However I will open a new PR to make rocm-cmake depend on rocm-core today and I will add entries into the ChangeLog for both these changes.

**Pull request merged**